apps/asap-server/test/controllers/events.test.ts

```diff
@@ -498,57 +499,117 @@ describe('Event controller', () => {
+          description: { iv: 'This event will be good' },
+          startDate: { iv: '2021-02-23T19:32:00Z' },
+          startDateTimeZone: { iv: 'Europe/Lisbon' },
+          endDate: { iv: '2021-02-23T19:32:00Z' },
+          endDateTimeZone: { iv: 'Europe/Lisbon' },
+          calendar: { iv: [calendarId] },
+          status: { iv: 'confirmed' },
+          tags: { iv: [] },
+        })
+        .reply(200, restEvent);
+
+      await events.create(eventData);
+    });
+
+    test('Should throw when squidex return an error', async () => {
+      nock(config.baseUrl)
+        .post(`/api/content/${config.appName}/events?publish=true`)
+        .reply(404);
+
+      await expect(events.create(eventData)).rejects.toThrow();
+    });
+  });
+
+  describe('Update method', () => {
+    const eventId = 'event-id';
+    const eventData: Partial<Event> = {
+      tags: ['kubernetes'],
+      meetingLink: 'https://zweem.com',
     };
 
-    test('Should create or update the event', async () => {
+    test('Should update the event', async () => {
       nock(config.baseUrl)
-        .patch(
-          `/api/content/${config.appName}/events/${eventId}?publish=true`,
-          {
-            title: { iv: 'Event Tittle' },
-            description: { iv: 'This event will be good' },
-            startDate: { iv: '2021-02-23T19:32:00Z' },
-            startDateTimeZone: { iv: 'Europe/Lisbon' },
-            endDate: { iv: '2021-02-23T19:32:00Z' },
-            endDateTimeZone: { iv: 'Europe/Lisbon' },
-            calendar: { iv: [calendarId] },
-            eventLink: { iv: 'event-link' },
-            status: { iv: 'confirmed' },
-            tags: { iv: [] },
-            meetingLink: { iv: 'https://meetings.com' },
-          },
-        )
-        .reply(200, eventResponse);
+        .patch(`/api/content/${config.appName}/events/${eventId}`, {
+          tags: { iv: ['kubernetes'] },
+          meetingLink: { iv: 'https://zweem.com' },
+        })
+        .reply(200, restEvent);
+
+      const result = await events.update(eventId, eventData);
+      expect(result).toStrictEqual(restEvent);
+    });
+
+    test('Should throw when squidex return an error', async () => {
+      nock(config.baseUrl)
+        .patch(`/api/content/${config.appName}/events/${eventId}`)
+        .reply(404);
+
+      await expect(events.update(eventId, eventData)).rejects.toThrow();
+    });
+  });
+
+  describe('fetchByGoogleId method', () => {
+    const googleId = 'google-event-id';
+    const filter = `data/googleId/iv eq '${googleId}'`;
+
+    test('Should throw when gets an error from squidex', async () => {
+      nock(config.baseUrl)
+        .get(`/api/content/${config.appName}/events`)
+        .query({
+          $top: 1,
+          $filter: filter,
+        })
+        .reply(500);
+
+      await expect(events.fetchByGoogleId(googleId)).rejects.toThrow();
+    });
+
+    test('Should return null when squidex returns an empty array', async () => {
```

Comment:
shouldn't these two methods behave the same when the event isn't found?

https://github.com/yldio/asap-hub/pull/691/files#diff-31921fb03563a6ca9a330c07a76c3cb5b9d69de98cd20c01997542447353f47bR379